src/main/scala/viper/gobra/backend/ViperBackends.scala

```diff
@@ -40,72 +40,56 @@ object ViperBackends {
     }
   }
 
-  object ViperServerBackend  {
-    var executor: VerificationExecutionContext = _
-    var server: ViperCoreServer = _
+  abstract class ViperServerBackend(initialServer: Option[ViperCoreServer] = None) extends ViperBackend {
+    private var server: Option[ViperCoreServer] = initialServer
 
-    def setExecutor(executionContext: VerificationExecutionContext): Unit = {
-      require(executor == null || executor.executorService.isShutdown)
-      executor = executionContext
-      // reset server since it depends on the execution context
-      resetServer()
-    }
-
-    def initServer(config: Config): Unit = {
-      require(server == null, "ViperCoreServer is already set.")
-      require(executor != null, "Executor is not set.")
+    /** should be implemented by classes implementing this abstract class */
```

Comment:
This comment could be shortened to `/* abstract method */`. 

Additionally, I would add a short comment to the methods in this class describing their intended use, specially in the abstract method (`getViperVerifierConfig`). This would be useful for future reference.